NoDefaultServletTest cannot find property quarkus.camel.servlet.url-patterns

```
[INFO] Running org.apache.camel.quarkus.component.servlet.test.NoDefaultServletTest
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.179 s <<< FAILURE! - in org.apache.camel.quarkus.component.servlet.test.NoDefaultServletTest
[ERROR] org.apache.camel.quarkus.component.servlet.test.NoDefaultServletTest  Time elapsed: 0.179 s  <<< ERROR!
java.lang.RuntimeException: java.lang.RuntimeException: java.util.NoSuchElementException: Property quarkus.camel.servlet.url-patterns not found
Caused by: java.lang.RuntimeException: java.util.NoSuchElementException: Property quarkus.camel.servlet.url-patterns not found
Caused by: java.util.NoSuchElementException: Property quarkus.camel.servlet.url-patterns not found
```

--- a/extensions/servlet/deployment/src/main/java/org/apache/camel/quarkus/component/servlet/deployment/ServletProcessor.java
+++ b/extensions/servlet/deployment/src/main/java/org/apache/camel/quarkus/component/servlet/deployment/ServletProcessor.java
@@ -16,7 +16,9 @@
  */
 package org.apache.camel.quarkus.component.servlet.deployment;
 
+import java.util.List;
 import java.util.Map.Entry;
+import java.util.Optional;
 
 import io.quarkus.arc.deployment.AdditionalBeanBuildItem;
 import io.quarkus.deployment.annotations.BuildProducer;
@@ -66,7 +68,8 @@ class ServletProcessor {
     static ServletBuildItem newServlet(String key, ServletConfig servletConfig,
             BuildProducer<AdditionalBeanBuildItem> additionalBean) {
         final String servletName = servletConfig.getEffectiveServletName(key);
-        if (servletConfig.urlPatterns.isEmpty()) {
+        final Optional<List<String>> urlPatterns = servletConfig.urlPatterns;
+        if (!urlPatterns.isPresent() || urlPatterns.get().isEmpty()) {
             throw new IllegalStateException(
                     String.format("Missing quarkus.camel.servlet%s.url-patterns",
                             ServletConfig.DEFAULT_SERVLET_NAME.equals(servletName) ? "" : "." + servletName));
@@ -74,7 +77,7 @@ class ServletProcessor {
 
         final Builder builder = ServletBuildItem.builder(servletName, servletConfig.servletClass);
         additionalBean.produce(new AdditionalBeanBuildItem(servletConfig.servletClass));
-        for (String pattern : servletConfig.urlPatterns) {
+        for (String pattern : urlPatterns.get()) {
             builder.addMapping(pattern);
         }
 

--- a/extensions/servlet/deployment/src/test/java/org/apache/camel/quarkus/component/servlet/test/CustomDefaultServletClassTest.java
+++ b/extensions/servlet/deployment/src/test/java/org/apache/camel/quarkus/component/servlet/test/CustomDefaultServletClassTest.java
@@ -29,16 +29,15 @@ import org.jboss.shrinkwrap.api.ShrinkWrap;
 import org.jboss.shrinkwrap.api.asset.Asset;
 import org.jboss.shrinkwrap.api.asset.StringAsset;
 import org.jboss.shrinkwrap.api.spec.JavaArchive;
-import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.RegisterExtension;
 
-@Disabled("https://github.com/apache/camel-quarkus/issues/854")
 public class CustomDefaultServletClassTest {
     @RegisterExtension
     static final QuarkusUnitTest CONFIG = new QuarkusUnitTest()
             .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
                     .addClass(Routes.class)
+                    .addClass(CustomServlet.class)
                     .addAsResource(applicationProperties(), "application.properties"));
 
     @Test

--- a/extensions/servlet/deployment/src/test/java/org/apache/camel/quarkus/component/servlet/test/NoDefaultServletTest.java
+++ b/extensions/servlet/deployment/src/test/java/org/apache/camel/quarkus/component/servlet/test/NoDefaultServletTest.java
@@ -29,15 +29,14 @@ import org.jboss.shrinkwrap.api.ShrinkWrap;
 import org.jboss.shrinkwrap.api.asset.Asset;
 import org.jboss.shrinkwrap.api.asset.StringAsset;
 import org.jboss.shrinkwrap.api.spec.JavaArchive;
-import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.RegisterExtension;
 
-@Disabled("https://github.com/apache/camel-quarkus/issues/853")
 public class NoDefaultServletTest {
     @RegisterExtension
     static final QuarkusUnitTest CONFIG = new QuarkusUnitTest()
             .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addClass(CustomServlet.class)
                     .addClass(CustomDefaultServletClassTest.Routes.class)
                     .addAsResource(applicationProperties(), "application.properties"));
 

--- a/extensions/servlet/runtime/src/main/java/org/apache/camel/quarkus/servlet/runtime/CamelServletConfig.java
+++ b/extensions/servlet/runtime/src/main/java/org/apache/camel/quarkus/servlet/runtime/CamelServletConfig.java
@@ -18,6 +18,7 @@ package org.apache.camel.quarkus.servlet.runtime;
 
 import java.util.List;
 import java.util.Map;
+import java.util.Optional;
 
 import io.quarkus.runtime.annotations.ConfigGroup;
 import io.quarkus.runtime.annotations.ConfigItem;
@@ -58,7 +59,7 @@ public final class CamelServletConfig {
          * patterns: {@code /*}, {@code /services/*}
          */
         @ConfigItem
-        public List<String> urlPatterns;
+        public Optional<List<String>> urlPatterns;
 
         /** A fully qualified name of a servlet class to serve paths that match {@link #urlPatterns} */
         @ConfigItem(defaultValue = DEFAULT_SERVLET_CLASS)
@@ -77,7 +78,7 @@ public final class CamelServletConfig {
          *         default values. Otherwise returns {@code false}.
          */
         public boolean isValid() {
-            return !urlPatterns.isEmpty();
+            return urlPatterns.isPresent() && !urlPatterns.get().isEmpty();
         }
 
         /**

